# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: docshell/base/BrowsingContext.cpp
# Commit: 771cac79ef3a
# Full Hash: 771cac79ef3a1f329b8a9140bd4533bee2555cca
# Author: Matt Woodrow <mwoodrow@mozilla.com>
# Date: 2019-12-12 09:53:26
# Description:
#   Bug 1602366 - Don't try to close if we're already discarded. r=mccr8
#   
#   Depends on D56820
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D56821
# ==============================================================================

diff -r ddff1be81b6c -r 771cac79ef3a docshell/base/BrowsingContext.cpp
--- a/docshell/base/BrowsingContext.cpp	Thu Dec 12 02:32:00 2019 +0000
+++ b/docshell/base/BrowsingContext.cpp	Thu Dec 12 02:31:55 2019 +0000
@@ -990,6 +990,9 @@
 }
 
 void BrowsingContext::Close(CallerType aCallerType, ErrorResult& aError) {
+  if (mIsDiscarded) {
+    return;
+  }
   // FIXME We need to set mClosed, but only once we're sending the
   //       DOMWindowClose event (which happens in the process where the
   //       document for this browsing context is loaded).
